### PR TITLE
Generate: `min_tokens_to_keep` has to be `>= 1`

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

As pointed out by @njhill in [this comment](https://github.com/huggingface/transformers/pull/24111#issuecomment-1601824441), `min_tokens_to_keep` has to be `>=1`. When it is not, the sampling step will lead to numerical exceptions, as all tokens have `-float("inf")` as logits.

This PR updates some of the checks, which were checking that it was `>=0`, and fixes the typical_p logits processor, which has the exact same issue as the one fixed in https://github.com/huggingface/transformers/pull/24111

### CODE REVIEW

File: src/transformers/generation/logits_process.py [hunk 1]
```diff
@@ -344,9 +346,7 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
         last_ind = (cumulative_probs < self.mass).sum(dim=1)
         last_ind[last_ind < 0] = 0
         sorted_indices_to_remove = sorted_scores > sorted_scores.gather(1, last_ind.view(-1, 1))
-        if self.min_tokens_to_keep > 1:
-            # Keep at least min_tokens_to_keep (set to min_tokens_to_keep-1 because we add the first one below)
-            sorted_indices_to_remove[..., : self.min_tokens_to_keep] = 0
+        sorted_indices_to_remove[..., : self.min_tokens_to_keep] = 0
```
Comment (gante) [inline src/transformers/generation/logits_process.py:349]: This change is the same as in https://github.com/huggingface/transformers/pull/24111/files

**Review by sgugger — APPROVED** [review]
Thanks for the fixes!

**Comment by njhill** [comment]
Thanks @gante!
